### PR TITLE
api: Merge profiles in batches of configurable size

### PR DESCRIPTION
**Pull request by brancz**

### CODE REVIEW

File: all.go [hunk 1]
```diff
@@ -36,13 +38,39 @@ func registerAll(m map[string]setupFunc, app *kingpin.Application, name string,
 	configFile := cmd.Flag("config.file", "Config file to use.").
 		Default("conprof.yaml").String()
 	retention := extkingpin.ModelDuration(cmd.Flag("storage.tsdb.retention.time", "How long to retain raw samples on local storage. 0d - disables this retention").Default("15d"))
+	maxMergeBatchSize := cmd.Flag("max-merge-batch-size", "Bytes loaded in one batch for merging. This is to limit the amount of memory a merge query can use. Defaults to 64Mb batch size.").
```
Comment (roidelapluie) [inline all.go:41]: Do you want to print the default? It would be displayed by kingpin.

File: all.go [hunk 2]
```diff
@@ -36,13 +38,39 @@ func registerAll(m map[string]setupFunc, app *kingpin.Application, name string,
...
+			g,
+			mux,
+			probe,
+			reg,
+			logger,
+			*storagePath,
+			*configFile,
+			*retention,
+			reloadCh,
+			*maxMergeBatchSize,
```
Comment (roidelapluie) [inline all.go:56]: At some point this could be passed as a struct for readability I think

File: api.go [hunk 3]
```diff
@@ -39,6 +41,8 @@ func registerApi(m map[string]setupFunc, app *kingpin.Application, name string)
 
 	storeAddress := cmd.Flag("store", "Address of statically configured store.").
 		Default("127.0.0.1:10901").String()
+	maxMergeBatchSize := cmd.Flag("max-merge-batch-size", "Bytes loaded in one batch for merging. This is to limit the amount of memory a merge query can use. Defaults to 64Mb batch size.").
```
Comment (roidelapluie) [inline api.go:44]: same comment

File: api.go [hunk 4]
```diff
@@ -39,6 +41,8 @@ func registerApi(m map[string]setupFunc, app *kingpin.Application, name string)
...
+		Default(strconv.Itoa(1024 * 1024 * 64)).Uint64()
```
Comment (roidelapluie) [inline api.go:45]:
```suggestion
		Default(strconv.Itoa(1024 * 1024 * 64)).Bytes()
```

Comment (brancz) [reply api.go:45]: Ah nice, I didn't see that one!

**Pull request merged**